src/py42/modules/securitydata.py

```diff
@@ -169,6 +177,71 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response["fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        response = file_event_client.get_file_location_detail_by_sha256(sha256_hash)
+
+        for device_id, paths in _parse_file_location_response(response):
+            try:
+                yield pds_client.find_file_versions(md5_hash, sha256_hash, device_id, paths)
+            except Py42HTTPError:
```

Comment:
Could you add a comment here explaining why we have to ignore all HTTPErrors here? Seems odd, which if the params are bad?

Reply:
@kiran-chaudhary, I thought we discussed this previously and agreed that we would log a message if an exception occurs?

Reply:
we should also do similar logging in `_stream_file()`.

Reply:
Done.